src/android/SupportKitCordova.java

```diff
@@ -0,0 +1,129 @@
+package io.supportkit.cordova;
+
+import io.supportkit.core.*;
+import io.supportkit.ui.ConversationActivity;
+
+import org.apache.cordova.CordovaPlugin;
+import org.apache.cordova.CallbackContext;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Date;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class SupportKitCordova extends CordovaPlugin {
+
+	@Override
+    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
+        if (action.equals("init")) {
+            callbackContext.error("Initialize must be done from the main application");
+            return false;
+        } else if (action.equals("show") || action.equals("showConversation")) {
+            this.show(callbackContext);
+        } else if (action.equals("setUser")) {
+            this.setUser(args, callbackContext);
+        } else if (action.equals("setUserProperties")) {
+            this.setUserProperties(args, callbackContext);
+        } else if (action.equals("track")) {
+            this.track(args, callbackContext);
+        } else {
+        	callbackContext.error("SupportKit method not supported");
+        	return false;
+        }
+
+        return true;
+    }
+
+    private void show(CallbackContext callbackContext) {
+        ConversationActivity.show(this.cordova.getActivity());
+        callbackContext.success();
+    }
+
+    private void setUser(JSONArray args, CallbackContext callbackContext) {
+        try {
```

Comment:
Instead of doing User.getCurrentUser() every time, create a variable here final User user = User.getCurrentUser()